numba/tests/test_operators.py

```diff
@@ -1518,10 +1518,10 @@ def test_impl2(a):
             return a == b_val
 
         def test_impl3():
-            r1 = True == True
-            r2 = True == False
-            r3 = False == True
-            r4 = False == False
+            r1 = True is True
```

Comment:
I think this is incorrect, `operator.eq` is being tested not `operator.is_` (same for the change below with `s/!=/is not/`).

Reply:
Since there will be no place in the code comparing to a singleton like True/False/None, then this test can be removed?

Reply:
There will indeed be no place in _Numba's_ code base that does this. But Numba is a compiler and the input space to the compiler is "literally any valid Python code a user can write", regardless of whether it is `flake8` compliant or even compilable by Numba.

Whilst it might be desirable for input code to not contain things like `operator.eq` on booleans, in practice, it probably does, and whilst breaking support for this might enforce code bases into `flake8` compliance, it'd also potentially break code already in production. Further, part of Numba's aim is to replicate the Python language semantics and `operator.eq(boolean, boolean)` is  valid Python.

As a result, I think this test and implementation has to stay as-is for as long as it is legal in Python to do e.g. `True == True`, as the test is checking that Numba can correctly compile this expression.

Reply:
I've removed these diff changes from the `Manually fix additional Flake8 E7 issues not fixed by autopep8` commit and updated `Enforce Flake8 E7 statement rules` to include the inline exception for E712 in these specific cases.